mongo/session.go

```diff
@@ -193,9 +194,12 @@ func (s *sessionImpl) WithTransaction(ctx context.Context, fn func(sessCtx Sessi
 			default:
 			}
 
-			if cerr, ok := err.(CommandError); ok {
-				if cerr.HasErrorLabel(driver.TransientTransactionError) {
-					continue
+			currErr := err
+			for ; currErr != nil; currErr = unwrap(currErr) {
+				if cerr, ok := currErr.(CommandError); ok {
```

Comment:
nit: You can reduce the nesting level by merging this and the check below into `ok && cerr.HasErrorLabel(...)`.

Reply:
code removed